File: frontend/src/app/models/vacancy/vacancy.ts
export interface Vacancy {
  id: number;
  organizationId: number;
  organizationName: string | null;
  organizationLogo: string | null;
  title: string;
  description: string;
  profession: string;
  industry: string;
  employmentType: string;
  employmentTypeName: string;
  experienceLevel: string;
  experienceLevelName: string;
  salaryMin?: number;
  salaryMax?: number;
  salaryCurrency?: string;
  salaryRange?: string;
  location: string;
  isRemote: boolean;
  status: string;
  statusName: string;
  applicationDeadline: string;
  publishedAt?: string | null;
  viewCount: number;
  applicationCount: number;
  createdAt: string;
  testId?: number | null;
}

export interface VacancyDetails extends Vacancy {
  requirements: string;
  responsibilities: string;
  organizationWebsite?: string | null;
  createdBy: number;
  createdByName: string;
  updatedAt?: string | null;
  statistics?: VacancyStatistics;
  tests: VacancyTest[];
  hasApplied?: boolean;
}

export interface VacancyStatistics {
  viewCount: number;
  totalApplications: number;
  pendingApplications: number;
  reviewedApplications: number;
  shortlistedApplications: number;
  rejectedApplications: number;
  testsAssigned: number;
  testsCompleted: number;
  averageTestScore: number;
  interviewsScheduled: number;
  interviewsCompleted: number;
  daysActive: number;
  daysRemaining: number;
}

export interface VacancyTest {
  id: number;
  title: string;
  duration: number;
  questionCount: number;
  isActive: boolean;
}

export interface CreateVacancyRequest {
  title: string;
  organizationId: number;
  description: string;
  requirements: string;
  responsibilities: string;
  profession: string;
  industry: string;
  employmentType: string;
  experienceLevel: string;
  salaryMin?: number;
  salaryMax?: number;
  salaryCurrency: string;
  location: string;
  isRemote: boolean;
  status: string;
  applicationDeadline: Date;
}

export interface UpdateVacancyRequest extends CreateVacancyRequest {}

File: backend/Models/Recruitment/Vacancy.cs
using TalentBridge.Common.Entities;
using TalentBridge.Enums;
using TalentBridge.Enums.Recruitment;
using TalentBridge.Models.Analytics;
using TalentBridge.Models.Roles;

using TalentBridge.Models.Testing;

namespace TalentBridge.Models.Recruitment;

public class Vacancy : BaseEntity
{
    public int OrganizationId { get; set; }
    public Organization Organization { get; set; }
    
    public int CreatedByHRManagerId { get; set; } 
    public HRManager CreatedByHRManager { get; set; }
    
    public string Title { get; set; }
    public string Description { get; set; }
    public string Requirements { get; set; }
    public string Responsibilities { get; set; }
    public string Profession { get; set; }
    public string Industry { get; set; }
    
    public EMPLOYMENT_TYPE EmploymentType { get; set; }
    public EXPERIENCE_LEVEL ExperienceLevel { get; set; }
    
    public decimal? SalaryMin { get; set; }
    public decimal? SalaryMax { get; set; }
    public string SalaryCurrency { get; set; } = "GEL";
    
    public string Location { get; set; }
    public bool IsRemote { get; set; } = false;
    
    public VACANCY_STATUS Status { get; set; } = VACANCY_STATUS.Draft;
    public DateTime ApplicationDeadline { get; set; }
    public DateTime? PublishedAt { get; set; }
    public int ViewCount { get; set; } = 0;
    public bool IsDeleted { get; set; } = false;
    public DateTime? DeletedAt { get; set; }
    public int? TestId { get; set; }
    public Test? Test { get; set; }
    public List<Application>? Applications { get; set; }
    public List<SavedVacancy>? SavedByUsers { get; set; }
    public List<VacancyView>? Views { get; set; }
}